File: lib/url.js
export const DEFAULT_BASE_URL = 'https://bytifi.com'

export function normalizeBaseUrl(baseUrl) {
  return String(baseUrl || DEFAULT_BASE_URL).replace(/\/+$/, '')
}

export function validateBaseUrl(url) {
  const normalized = normalizeBaseUrl(url)

  let parsed
  try {
    parsed = new URL(normalized)
  } catch {
    throw new Error(`Invalid base URL: ${url}`)
  }

  if (!['http:', 'https:'].includes(parsed.protocol)) {
    throw new Error(`Invalid base URL protocol: ${parsed.protocol} (use http: or https:)`)
  }

  if (!parsed.hostname) {
    throw new Error('Invalid base URL: missing hostname.')
  }

  return normalized
}

File: python/src/bytifi/url.py
DEFAULT_BASE_URL = "https://bytifi.com"


def normalize_base_url(base_url: str | None) -> str:
    return str(base_url or DEFAULT_BASE_URL).rstrip("/")


def validate_base_url(url: str) -> str:
    from urllib.parse import urlparse

    normalized = normalize_base_url(url)
    parsed = urlparse(normalized)

    if parsed.scheme not in ("http", "https"):
        raise ValueError(f"Invalid base URL protocol: {parsed.scheme} (use http: or https:)")
    if not parsed.hostname:
        raise ValueError("Invalid base URL: missing hostname.")

    return normalized
